### DISCUSSION TITLE
Why does alembic put old tables in a new migration?

### DISCUSSION THREAD

**Discussion by Johnb6262** [sqlalchemy/alembic#1451] [Usage Questions]
I am setting up alembic in a new project.

I successfully got the first migration working in my local and aws postgres databases. Added 5 tables.

I then added a new model to the same .py file I had used before and ran the alembic command to make the migration file.

alembic revision --autogenerate -m "Add Message model"

I expected only the new table to show up in the migration file.  Instead, the new migration file contains all the old tables and the one new one I added for a total of 6.  

Of course, running the "alembic upgrade head" command fails because the old tables are already in my database.

sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DuplicateTable) relation "module" already exists

What have I misunderstood or done wrong?  I expected ONLY the new table to be picked up in the new migration file as with the Django migration tool I'm familiar with...

And - alembic writes ONLY the latest migration file's version number to the database?  How do I then track what has or hasn't happened in terms of migrations in that specific database?

**Comment by CaselIT** [comment]
Hi,

Can you clarify a bit how is your env setup?
Are you using schemas? If so how are those configured?

**Reply by CaselIT** [reply]
oh this should go in alembic. Migrating it there

**Reply by CaselIT** [reply]
@Johnb6262 ping since I don't know if you get a notification of the new location

**Reply by matildeRivas** [reply]
I have the same problem, were you able to fix it?
